plutus-streaming/src/Plutus/Streaming.hs

```diff
@@ -0,0 +1,127 @@
+module Plutus.Streaming
+  ( SimpleChainSyncEvent,
+    withSimpleChainSyncEventStream,
+    ChainSyncEvent (..),
+    ChainSyncEventException (..),
+  )
+where
+
+import Cardano.Api (BlockInMode, CardanoMode, ChainPoint, ChainSyncClient (ChainSyncClient), ChainTip,
+                    ConsensusModeParams (CardanoModeParams), EpochSlots (EpochSlots),
+                    LocalChainSyncClient (LocalChainSyncClient),
+                    LocalNodeClientProtocols (LocalNodeClientProtocols, localChainSyncClient, localStateQueryClient, localTxSubmissionClient),
+                    LocalNodeConnectInfo (LocalNodeConnectInfo, localConsensusModeParams, localNodeNetworkId, localNodeSocketPath),
+                    NetworkId, connectToLocalNode)
+import Cardano.Api.ChainSync.Client (ClientStIdle (SendMsgDone, SendMsgFindIntersect, SendMsgRequestNext),
+                                     ClientStIntersect (ClientStIntersect, recvMsgIntersectFound, recvMsgIntersectNotFound),
+                                     ClientStNext (ClientStNext, recvMsgRollBackward, recvMsgRollForward))
+import Control.Concurrent.Async (withAsync)
+import Control.Concurrent.STM (TChan, atomically, dupTChan, newBroadcastTChanIO, readTChan, writeTChan)
+import Control.Exception (Exception, throw)
+import GHC.Generics (Generic)
+import Streaming (Of, Stream)
+import Streaming.Prelude qualified as S
+
+data ChainSyncEvent a
+  = RollForward a ChainTip
+  | RollBackward ChainPoint ChainTip
+  deriving (Show, Functor, Generic)
+
+type SimpleChainSyncEvent = ChainSyncEvent (BlockInMode CardanoMode)
+
+data ChainSyncEventException
+  = NoIntersectionFound
+  deriving (Show)
+
+instance Exception ChainSyncEventException
+
+withSimpleChainSyncEventStream ::
+  FilePath ->
+  NetworkId ->
+  -- | The point on the chain to start streaming from
+  ChainPoint ->
+  (Stream (Of SimpleChainSyncEvent) IO r -> IO b) ->
+  IO b
+withSimpleChainSyncEventStream socketPath networkId point consumer = do
+  -- The chain-sync client runs in a different thread and it will send us
+  -- block through this channel.
+  chan <- newBroadcastTChanIO
+  readerChannel <- atomically $ dupTChan chan
```

Comment:
Can you add a comment here why `dupTChan` is needed here. What happens if we just use `chan` instead?

Reply:
I added a note linking `newBroadcastTChan` [documentation](https://hackage.haskell.org/package/stm-2.5.0.2/docs/Control-Concurrent-STM-TChan.html#v:newBroadcastTChan).